components/constellation/constellation.rs

```diff
@@ -3209,9 +3214,12 @@ where
     fn handle_navigate_request(
         &self,
         id: PipelineId,
+        top_level_browsing_context_id: TopLevelBrowsingContextId,
         request_builder: RequestBuilder,
         cancel_chan: IpcReceiver<()>,
     ) {
+        let request_builder =
+            request_builder.target_browsing_context(Some(top_level_browsing_context_id));
```

Comment:
So I'm a bit concerned, because this means that resource loading (images, fonts, etc) are not associated with a WebView, but they probably should be. This requires a more thorough change though -- likely the TopLevelBrowsingContextId needs to be passed in as part of the `RequestBuilder::new` method.

Reply:
After more digging, I wonder if we should use `TopLevelBrowsingContextId` or `BrowsingContextId` here since we might load an iframe context. (or maybe both?)

Reply:
In the end, both, likely make sense, but `TopLevelBrowsingContextId` is what we need for now:
 - `TopLevelBrowsingContextId`: the WebView which can be used to associate the prompt dialog with a particular web view.
 - `BrowsingContextId`: the frame, which the embedder might want to have frame specific behavior. I'm not sure if we have a strong concept of a `Frame` in the API yet.